From 67ed1bdb1af1c70a5fd3b377a997401b7676b827 Mon Sep 17 00:00:00 2001
From: jaipaul Cheernam <jaipaul.cheernam@est.tech>
Date: Sat, 15 Aug 2026 14:20:03 -0400
Subject: [PATCH] exportfs: link failure with --disable-nfsdctl

Building nfs-utils 2.9.2 with --disable-nfsdctl fails to link
exportfs. This is a regression from 2.9.1.

Signed-off-by: Steve Dickson <steved@redhat.com>
Upstream: https://git.linux-nfs.org/?p=steved/nfs-utils.git;a=commit;h=67ed1bdb1af1c70a5fd3b377a997401b7676b827
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4ea19715..2ce1e129 100644
--- a/configure.ac
+++ b/configure.ac
@@ -256,8 +256,6 @@ PKG_CHECK_MODULES(LIBNL3, libnl-3.0 >= 3.1)
 PKG_CHECK_MODULES(LIBNLGENL3, libnl-genl-3.0 >= 3.1)
 
 AC_CHECK_HEADERS(linux/nfsd_netlink.h)
-AC_DEFINE([HAVE_NFSD_NETLINK], 1,
-	  [Define to 1 if nfsd generic netlink support is available])
 
 # ensure the system netlink headers have the latest features
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <linux/nfsd_netlink.h>]],
@@ -279,6 +277,8 @@ AC_ARG_ENABLE(nfsdctl,
 	enable_nfsdctl="yes")
 	AM_CONDITIONAL(CONFIG_NFSDCTL, [test "$enable_nfsdctl" = "yes" ])
 	if test "$enable_nfsdctl" = yes; then
+		AC_DEFINE([HAVE_NFSD_NETLINK], 1,
+	  		[Define to 1 if nfsd generic netlink support is available])
 		PKG_CHECK_MODULES(LIBREADLINE, readline)
 	fi
 
-- 
2.53.0

